cpp/include/gar/graph.h

```diff
@@ -317,28 +322,28 @@ class EdgeIter {
    * @param index_converter The converter for transforming the edge chunk
    * indices.
    */
-  explicit EdgeIter(const EdgeInfo& edge_info, const std::string& prefix,
-                    AdjListType adj_list_type, IdType global_chunk_index,
-                    IdType offset, IdType chunk_begin, IdType chunk_end,
+  explicit EdgeIter(const std::shared_ptr<EdgeInfo>& edge_info,
```

Comment:
Hi, @acezen . I would like to understand when to use Reader and when to use shared_ptr<Reader>. In the code, I noticed that offsetReader in EdgeIter is a shared_ptr (since it is only validate for CSR/CSC), while others are instances.